kotest-extensions/kotest-extensions-testcontainers/src/jvmMain/kotlin/io/kotest/extensions/testcontainers/StartablePerTestListener.kt

```diff
@@ -35,4 +36,54 @@ class StartablePerTestListener<T : Startable>(val startable: T) : TestListener {
          startable.stop()
       }
    }
+
+   override suspend fun beforeContainer(testCase: TestCase) {
+      if (testCase.type != TestType.Container) return
+
+      withContext(Dispatchers.IO) {
+         testLifecycleAwareListener.beforeTest(testCase)
+         startable.start()
+      }
+   }
+
+   override suspend fun afterContainer(testCase: TestCase, result: TestResult) {
+      if (testCase.type != TestType.Container) return
+
+      withContext(Dispatchers.IO) {
+         testLifecycleAwareListener.afterTest(testCase, result)
+         startable.stop()
+      }
+   }
+
+   override suspend fun beforeEach(testCase: TestCase) {
+      if (testCase.type != TestType.Test) return
+
+      withContext(Dispatchers.IO) {
+         testLifecycleAwareListener.beforeTest(testCase)
+         startable.start()
+      }
+   }
+
+   override suspend fun afterEach(testCase: TestCase, result: TestResult) {
+      if (testCase.type != TestType.Test) return
+
+      withContext(Dispatchers.IO) {
+         testLifecycleAwareListener.afterTest(testCase, result)
+         startable.stop()
+      }
+   }
+
+   override suspend fun beforeAny(testCase: TestCase) {
+      withContext(Dispatchers.IO) {
+         testLifecycleAwareListener.beforeTest(testCase)
+         startable.start()
```

Comment:
This will call the .start() method 3 times, once for beforeTest, beforeAny and beforeEach / beforeContainer